go/arrow/type_traits_decimal128.go

```diff
@@ -39,8 +40,13 @@ func (decimal128Traits) BytesRequired(n int) int { return Decimal128SizeBytes *
 
 // PutValue
 func (decimal128Traits) PutValue(b []byte, v decimal128.Num) {
-	binary.LittleEndian.PutUint64(b[:8], uint64(v.LowBits()))
-	binary.LittleEndian.PutUint64(b[8:], uint64(v.HighBits()))
+	if runtime.GOARCH == "s390x" {
+		binary.BigEndian.PutUint64(b[:8], uint64(v.LowBits()))
```

Comment:
Can we use [this convention](https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/basic_decimal.cc#L145-L154) across languages? In other words, `LowBits()` always keeps lower 64-bits regardless of endian. `HighBits()` keeps higher 64-bits.

Reply:
Thanks for the review.  I'll fix this.